tests/console-backend-service/internal/domain/serverless/function_test.go

```diff
@@ -0,0 +1,308 @@
+	for _, function := range actual {
+		assert.Equal(t, expectedNamespace, function.Namespace)
+		assert.Equal(t, true, isInArray(function.Name, expectedNames))
+	}
+}
+
+func isInArray(data string, array []string) bool {
+	for _, elem := range array {
+		if elem == data {
+			return true
+		}
+	}
+	return false
+}
+
+func queryFunctions(client *graphql.Client, namespace, details string) ([]Function, error) {
+	query := fmt.Sprintf(`
+		query{
+			functions (
+				namespace: "%s"
+			){
+				%s
+			}
+		}
+	`, namespace, details)
+
+	req := graphql.NewRequest(query)
+	var res FunctionListQueryResponse
+	err := client.Do(req, &res)
+
+	return res.Functions, err
+}
+
+func checkFunctionQuery(t *testing.T, expected, actual Function) {
+	assert.Equal(t, expected.Name, actual.Name)
+	assert.Equal(t, expected.Namespace, actual.Namespace)
+	assert.Equal(t, len(expected.Labels), len(actual.Labels))
+}
+
+func checkFunctionEvent(t *testing.T, expected, actual FunctionEvent) {
+	assert.Equal(t, expected.Type, actual.Type)
+	checkFunctionQuery(t, expected.Function, actual.Function)
+}
+
+func fixFunctionEvent(eventType string, function Function) FunctionEvent {
+	return FunctionEvent{
+		Type:     eventType,
+		Function: function,
+	}
+}
+
+func fixFunction(nameSuffix, namespace string, labels []string) Function {
+	labelTemplate := map[string]string{}
+	for _, label := range labels {
+		labelTemplate[label] = label
+	}
+
+	return Function{
+		Name:      fmt.Sprintf("%s-%s", FunctionNamePrefix, nameSuffix),
+		Namespace: namespace,
+		Labels:    labelTemplate,
+	}
+}
+
+func readFunctionEvent(sub *graphql.Subscription) (FunctionEvent, error) {
+	type Response struct {
+		FunctionEvent FunctionEvent
+	}
+
+	var response Response
+	err := sub.Next(&response, tester.DefaultDeletionTimeout)
+
+	return response.FunctionEvent, err
+}
+
+func queryFunction(client *graphql.Client, arguments, details string) (Function, error) {
+	query := fmt.Sprintf(`
+		query{
+			function (
+				%s
+			){
+				%s
+			}
+		}
+	`, arguments, details)
+	type Response struct {
+		Function Function
+	}
+
+	req := graphql.NewRequest(query)
+	var res Response
+	err := client.Do(req, &res)
+
+	return res.Function, err
+}
+
+func mutationFunction(client *graphql.Client, requestType, arguments, resourceDetailsQuery string) error {
+	query := fmt.Sprintf(`
+		mutation {
+			%s (
+				%s
+			){
+				%s
+			}
+		}
+	`, requestType, arguments, resourceDetailsQuery)
+	req := graphql.NewRequest(query)
+	err := client.Do(req, nil)
+
+	return err
+}
+
+func queryFunctionArguments(nameSuffix, namespace string) string {
+	return fmt.Sprintf(`
+		name: "%s-%s",
+		namespace: "%s"
+	`, FunctionNamePrefix, nameSuffix, namespace)
+}
+
+func deleteManyFunctionsArguments(namespace string, functionsMetadata []FunctionMetadataInput) string {
+	functions := ""
+	for i, meta := range functionsMetadata {
+		functions += fmt.Sprintf(`{name: "%s", namespace: "%s"}`, meta.Name, meta.Namespace)
```

Comment:
You can use json.Marshal function for that logic.

Reply:
`json.Marschal` returns value and error, so I decided to create my own mechanism to convert the string array to the json string